data/lua/wml/kill.lua

```diff
@@ -3,6 +3,10 @@ local location_set = wesnoth.require "location_set"
 local kill_recursion_preventer = location_set.create()
 
 function wesnoth.wml_actions.kill(cfg)
+	if wml.get_child(cfg, "filter") then
+		wml.error "do not use [filter] in [kill]"
```

Comment:
The writing style doesn't match similar error messages that are already in the code. Suggestions:

* "[kill] has unsupported [filter] subtag"
* "[filter] tag found inside [kill]"

Reply:
`"[kill] has unsupported [filter] subtag"` looks nicer.

Reply:
schema uses 20230713 19:25:15 error validation: Tag [event] may not be used in [effect]

Reply:
That sounds good to me too.